src/pudl/analysis/allocate_net_gen.py

```diff
@@ -902,7 +895,12 @@ def _allocate_unassociated_records(
 
     eia_generators_unassociated = (
         gen_assoc[~connected_mask]
-        .assign(energy_source_code_num="energy_source_code_1")
+        .pipe(
+            pudl.transform.classes.drop_invalid_rows,
+            pudl.transform.classes.InvalidRows(
+                required_valid_cols=data_columns, invalid_values=[pd.NA, np.nan, 0.0]
+            ),
+        )
```

Comment:
~ half of all of these unconnected records were 0's or nulls. this whole function is here to ensure we  don't loose fuel in the association process. And 0 fuel isn't really fuel for this purpose. was easy to reuse the `drop_invalid_rows` transformer we made for ferc1 transformers

Reply:
I'm wondering if we do want to keep zero values here. A zero is still data about what was being consumed, and I'm wondering if this could be valuable. Have you tried running it both ways to see if the outputs differ (if you keep 0s in this step)?

Reply:
I removed them in this case because these are the unassociated records and later in _allocate_unassociated_data_col they will be added to the associated records. so adding a zero to a value will result in no change. _allocate_unassociated_data_col already does nothing if the unassociated value is null because you can't add a null to a value.

But I do think there is some merit in keeping the zero's in because we do the following: `og_value.fillna(0) + unassociated_value` so this will result in more 0's in the case of the `og_value` being null.